### ISSUE TITLE
Question about org-element vs. om

### ISSUE THREAD

**Issue by alphapapa** [ndw/org-to-xml#4]
Hi,

Quick question if I may: of course, Nate's `om` library is great.  But I'm curious, why do you use `om-parse-element-at` here: https://github.com/ndw/org-to-xml/blob/aefea2175e850c3c3d6a0fddc76d75d7e3bba662/om-to-xml.el#L200

Would `org-element-parse-buffer` be sufficient?  AFAIK `om` is more intended for manipulating the structures that `org-element` produces than doing the parsing itself.

BTW, very cool project, even though I kind of recoil at XML in general.  :)  Always glad to see more projects in the Org ecosystem.

**Comment by ndw** [comment]
I struggled mightily with decoding the `org-element-*` structures. Perhaps my elisp skills weren't up to the task, perhaps I failed to find the relevant documentation. If you look in `org-to-xml.el`, you'll see my crude attempts to work with the underlying org structures. It never worked very well and there were cases where it failed in odd ways.

The short answer is: I discovered the `om` library and concluded that it gave me more regular, more understandable structure to work with.

**Comment by alphapapa** [comment]
I sympathize, as I have also struggled with the `org-element` structures.  For some reason they seemed unnatural to me for a long time.  I even tried to use `treepy`'s zipper tools to work on them and failed (though maybe I should try that again since I've learned more).  

Eventually they sort of clicked with me.  Perhaps ironically, I think it was noticing how the structures returned by Emacs's libxml parsing functions are similar to the ones returned by `org-element` that helped me.  As well, I started using destructuring whenever I need to work with them, e.g. in [these functions](https://github.com/alphapapa/org-graph-view/blob/13314338d70d2c19511efccc491bed3ca0758170/org-graph-view.el#L335), which makes it much easier for me to keep the elements in my head than using `car`/`cdr`-type accessors.

> The short answer is: I discovered the om library and concluded that it gave me more regular, more understandable structure to work with.

That's what confuses me: I thought that the structures `om` uses are the same `org-element` structures, but maybe I'm wrong.

Thanks.
